crates/oxc_linter/src/rules/jest/prefer_mock_promise_shorthand.rs

```diff
@@ -0,0 +1,441 @@
+use oxc_ast::{
+    ast::{Argument, CallExpression, Expression, Statement},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, CompactStr, Span};
+
+use crate::{context::LintContext, fixer::Fix, rule::Rule, utils::get_node_name};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jest(prefer-mock-promise-shorthand): Prefer mock resolved/rejected shorthands for promises")]
+#[diagnostic(severity(warning), help("Prefer {0:?}"))]
+struct UseMockShorthand(CompactStr, #[label] Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferMockPromiseShorthand;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// When working with mocks of functions that return promises, Jest provides some
+    /// API sugar functions to reduce the amount of boilerplate you have to write.
+    /// These methods should be preferred when possible.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// // invalid
+    /// jest.fn().mockImplementation(() => Promise.resolve(123));
+    /// jest
+    ///   .spyOn(fs.promises, 'readFile')
+    ///   .mockReturnValue(Promise.reject(new Error('oh noes!')));
+    ///
+    /// myFunction
+    ///   .mockReturnValueOnce(Promise.resolve(42))
+    ///   .mockImplementationOnce(() => Promise.resolve(42))
+    ///   .mockReturnValue(Promise.reject(new Error('too many calls!')));
+    ///
+    /// // valid
+    /// ```javascript
+    /// jest.fn().mockResolvedValue(123);
+    /// jest.spyOn(fs.promises, 'readFile').mockRejectedValue(new Error('oh noes!'));
+    ///
+    /// myFunction
+    ///   .mockResolvedValueOnce(42)
+    ///   .mockResolvedValueOnce(42)
+    ///   .mockRejectedValue(new Error('too many calls!'));
+    /// ```
+    ///
+    PreferMockPromiseShorthand,
+    correctness,
```

Comment:
The described is "Jest provides some API sugar functions to reduce the amount of boilerplate you have to write.", maybe ["style"](https://doc.rust-lang.org/nightly/clippy/) is better?